params: make userID a query parameter in /v1/uid endpoints

mhilton opened the pull request:
The userID is intended to be a system-wide unique value for the user,
separate from username. Such values may be comprised partly from URLs
this makes them hard to process as a path segment.

jujugui commented:
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/30/

fabricematrat commented:
:shipit:

jujugui commented:
Status: merge request accepted. Url: http://ci-gce.theblues.io:8080/job/candid-merge

mhilton committed:
params: make userID a query parameter in /v1/uid endpoints

The userID is intended to be a system-wide unique value for the user,
separate from username. Such values may be comprised partly from URLs
this makes them hard to process as a path segment.
